### PR TITLE
Simplify Toggle Switch, add support for dark mode

### PR DESCRIPTION
**Pull request by stevekinney**
- Simplifies the code for the toggle switch.
- Adds support for dark mode.

## Checklist

- [x] Approval from @tigernaut

### CODE REVIEW

File: src/lib/holocene/toggle-switch.svelte [hunk 1]
```diff
@@ -1,129 +1,34 @@
...
   />
   <span
-    on:mouseenter={handleMouseEnter}
-    on:mouseleave={handleMouseLeave}
-    class="slider"
-    class:disabled
-    class:checked
-    class:focused
-    class:hovered
+    class="surface-subtle not-sr-only flex h-5 w-9 items-center rounded-xl border border-transparent p-px text-primary outline-2 outline-offset-1 outline-brand/50 group-hover:border-brand peer-checked:justify-end peer-checked:bg-brand peer-checked:text-inverse peer-focus:outline peer-disabled:opacity-50"
```
Comment (rossedfort) [inline src/lib/holocene/toggle-switch.svelte:30]: What's `p-px` ?

**Review by rossedfort — APPROVED** [review]

File: src/lib/holocene/toggle-switch.svelte (hunk 1 above)
Comment (rossedfort) [reply src/lib/holocene/toggle-switch.svelte:30]: Also I noticed there's no hover state when the toggle is "checked", which would be a regression. Were the color changes necessary for dark mode?

Comment (rossedfort) [reply src/lib/holocene/toggle-switch.svelte:30]: (Asking cause I followed [the designs](https://www.figma.com/file/LNN3Cu8Akqb240WmJmbtsS/Holocene?type=design&node-id=3177-74002&mode=design&t=TdMPLYu9vYeUA9TV-4) when I implemented this)

Comment (GraceGardner) [reply src/lib/holocene/toggle-switch.svelte:30]: @rossedfort Heyo, I merged this so I could use something from the plugin.ts that Steve and I were pairing on. I'll go through and double check this matches the new designs and have you review it later today or early tomorrow if that's cool?

Comment (rossedfort) [reply src/lib/holocene/toggle-switch.svelte:30]: I think it's OK as is if Ash signed off on it!

Comment (rossedfort) [reply src/lib/holocene/toggle-switch.svelte:30]: Just wanted to call attention to the changes

**Pull request merged**